### DISCUSSION TITLE
File and Folder renames should always update relevant links

### DISCUSSION THREAD

**Discussion by vHanda** [GitJournal/GitJournal#821] [Feature Requests]
Notes can link to other notes via a relative URL. In the case when a note/folder is moved or renamed it would be awesome if the notes which contained the relative links were also accordingly updated, so we do not result in broken links.

---
> Please vote on this issue with a 👍reaction. It's one of the primary mechanisms used to prioritize issues.

**Comment by aisbergde** [comment]
not always, but I should be informed
The main issue is: it should be possible to changes notes outside GitJounal (rename files, move to other folders). GitJournal would not be able to track this.

I would prefer UUID in notes metadata and to be able to define links using UUID, then renaming or moving will work. Of course that means you need to parse all notes for their UUID and to store this information in a mapping table/file. And it should be possible to reparse on request or automatically in some situtation. For example if there are changes. There was another issue to implement git logging, this can be used as a trigger to recreate the mapping table.
